odbc/src/api/bind_col_tests.rs

```diff
@@ -466,4 +354,478 @@ mod unit {
             let _ = Box::from_raw(buffer as *mut WChar);
         }
     }
+
+    #[test]
+    fn test_binding_arrays_when_rowset_size_evenly_divides_resultset_size() {
```

Comment:
I'll add this to my ticket regarding adding more test cases instead of doing it as part of this PR. I think it's better this way since there will probably be more refactor opportunities whenever I add more test cases anyways.